Rename 'Test Cluster' to 'Unit Testing' to unstutter class and item naming

andy31415 opened the pull request:
Suffixing things with 'Cluster' in codegen makes things stutter as `Cluster` is a suffix appended to many things.

This generally converts `TestClusterCluster` to just say `UnitTestingCluster`.

bzbarsky-apple commented on src/darwin/Framework/CHIP/zap-generated/MTRBaseClusters.h:
```diff
@@ -21063,89 +21063,84 @@ labels.
  *    The Test Cluster is meant to validate the generated code
  */
-@interface MTRBaseClusterTestCluster : MTRCluster
+@interface MTRBaseClusterTest : MTRCluster
```
This needs a backwards-compat shim, I expect; changing shipped Darwin public API without a deprecation period is generally not allowed...

This concern applies to the changes in MTRBaseClusters.h, MTRClusters.h, MTRCommandPayloadsObjc.h, and MTRStructsObjc.h

andy31415 replied:
@bzbarsky-apple  What do I need to do to have this covered?

Given that this is a "Cluster Test" that is not a real cluster, is this really to be considered a compatibility issue? I would not expect anyone to implement this for real, it should be only for our own all cluster apps.

bzbarsky-apple replied:
> Given that this is a "Cluster Test" that is not a real cluster, is this really to be considered a compatibility issue?

I checked internally, and yes.  This API shipped as a public API in Matter.framework, and any changes to it need to come with deprecation periods, etc.

As far as this being covered, we need a plan for how we will do this change without breaking the APIs involved.  It might be doable with just changes to the templates, or it might need ZAP-side changes.  I can probably look into what's needed there, but not until next week; this week the member meeting is taking up too much time.

If you're willing to wait until then, I can make sure I create a PR against this one that addresses the Darwin bits.

andy31415 replied:
Could we iterate on it to not block this PR too long? This PR is me yak-shaving to make cluster_id constants the same with cluster name and I change this because `test_cluster_cluster_id` sounded strange.

My thinking regarding limited-lifespan shims is that they should be in a separate file, maybe not driven by code generation and instead manually maintained. Not sure how to organize that as I am not familiar with how darwin code is organized.

andy31415 replied:
Resolving based on offline discussion: fixing darwin seems to take longer and will be done separately.

bzbarsky-apple commented:
> In hindsight, naming this Testing Validation or Unit Testing may have made it more unique.

If we're changing it _anyway_, maybe we should in fact do that....

andy31415 commented:
> > In hindsight, naming this Testing Validation or Unit Testing may have made it more unique.
> 
> If we're changing it _anyway_, maybe we should in fact do that....

On, renamed to `Unit Testing`. This also changed the defines (UNIT_TESTING_CLUSTER_ID and such)
